src/rust/ensogl/lib/list-view/src/entry.rs

```diff
@@ -114,6 +120,99 @@ impl ModelProvider for EmptyProvider {
 }
 
 
+// === Model Provider for Vectors ===
+
+/// An Entry Model Provider that wraps a Vec<T:Into<Model>>.
+#[derive(Clone,Debug)]
+pub struct VectorProvider<T> {
+    content : Vec<T>,
+}
+
+impl<T:Into<Model> + Debug + Clone> ModelProvider for VectorProvider<T> {
+    fn entry_count(&self) -> usize {
+        self.content.len()
+    }
+
+    fn get(&self, ix:usize) -> Option<Model> {
+        let item        = self.content.get(ix)?.clone();
+        let model:Model = item.into();
+        Some(model)
+    }
+}
+
+impl<T> From<Vec<T>> for VectorProvider<T> {
+    fn from(content: Vec<T>) -> Self {
+        VectorProvider{content}
+    }
+}
+
+
+// === Masked Model Provider ===
+
+/// An Entry Model Provider that wraps a Vec<T:Into<Model>> and allows the masking of a single item.
+#[derive(Clone,Debug)]
+pub struct SingleMaskedProvider {
```

Comment:
At this point I would consider putting all providers and trait into `provider` submodule, and rename everything here removing `Provider` suffix. Then we could reimport "ModelProvider" in entry.rs

Reply:
I would leave that for later to avoid conflicts with the searcher PR.

Reply:
The searcher PR landed, so it can be refactored in this one.